src/main/java/trackr/logic/parser/AddTaskCommandParser.java

```diff
@@ -0,0 +1,59 @@
+package trackr.logic.parser;
+
+import static trackr.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static trackr.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static trackr.logic.parser.CliSyntax.PREFIX_NAME;
+import static trackr.logic.parser.CliSyntax.PREFIX_STATUS;
+
+import java.util.stream.Stream;
+
+import trackr.logic.commands.AddTaskCommand;
+import trackr.logic.parser.exceptions.ParseException;
+import trackr.model.task.Task;
+import trackr.model.task.TaskDeadline;
+import trackr.model.task.TaskName;
+import trackr.model.task.TaskStatus;
+
+/**
+ * Parses input arguments and creates a new AddTaskCommand object
+ */
+
+public class AddTaskCommandParser implements Parser<AddTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddTaskCommand
+     * and returns an AddTaskCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddTaskCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DEADLINE, PREFIX_STATUS);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DEADLINE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTaskCommand.MESSAGE_USAGE));
+        }
+
+        TaskName taskName = ParserUtil.parseTaskName(argMultimap.getValue(PREFIX_NAME).get());
+        TaskDeadline taskDeadline = ParserUtil.parseTaskDeadline(argMultimap.getValue(PREFIX_DEADLINE).get());
+        TaskStatus taskStatus;
+        if (arePrefixesPresent(argMultimap, PREFIX_STATUS)) {
+            taskStatus = ParserUtil.parseTaskStatus(argMultimap.getValue(PREFIX_STATUS).get());
+        } else {
+            taskStatus = new TaskStatus("N");
+        }
```

Comment:
I feel like this could be moved towards the `ParserUtil` instead.

Maybe consider this function in ParserUtil.
```
    public static Tag parseStatus(Optional<String> taskStatus) throws ParseException {
        requireNonNull(taskStatus);
        if (!taskStatus.isPresent()) {
            return new TaskStatus();
        }

        String trimmedTaskStatus = taskStatus.get().trim();
        if (!TaskStatus.isValidTaskStatus(trimmedTaskStatus)) {
            throw new ParseException(TaskStatus.MESSAGE_CONSTRAINTS);
        }
        return new TaskStatus(trimmedTaskStatus);
    }
```